doc/sphinx-guides/source/api/native-api.rst

```diff
@@ -3976,6 +3976,64 @@ The fully expanded example above (without environment variables) looks like this
 
   curl "https://demo.dataverse.org/api/info/settings/:MaxEmbargoDurationInMonths"
 
+Get Exposed Settings
+~~~~~~~~~~~~~~~~~~~~
+
+Although the previous endpoints allow to get public settings individually, this endpoint unifies all public settings by exposing them individually or in groups.
+
+For supporting this, the public settings have been grouped considering different Dataverse scopes. Within these groups, the settings have been named in a natural way for the user, being this natural name independent of the setting source name on which they depend (JVM options or database settings).
+
+Groups follow a tree structure, starting with the root group and its root-level settings, and each child group can contain one or more settings and/or one or more child groups.
+
+Currently under the root group we find the following settings:
+
+- fqdn
+- isPublicInstall
+
+And the following child groups, with their associated settings:
+
+- api: termsOfUse, allowIncompleteMetadata.
+- dataset: publishPopupCustomText, allowedCurationLabels, zipDownloadLimit.
+- datafile: maxEmbargoDurationInMonths.
+
+The tree-like structure translates to the endpoint structure, which will receive child group names and setting names via path parameters, ordered by depth starting from root.
```

Comment:
Is there any performance concern? Could this lead to a denial of service attack?

Reply:
The root SettingGroup object containing all settings and subgroups is instantiated as a singleton, so reading the settings from all the sources, which is what I think could cause loading issues if done repeatedly, should be done once.

I have added an optimization including thread safety to the singleton instantiation.